### ISSUE TITLE
FE SPIKE: Resources & Support page shows incorrect breadcrumbs

### ISSUE THREAD

**Issue by jilladams** [department-of-veterans-affairs/va.gov-cms#10308]
## Describe the defect
https://dsva.slack.com/archives/CDHBKAL9W/p1660661901076319

The breadcrumbs at the top of this R&S article are showing incorrectly. It should be Home > Resources and support > Choosing a decision review option. But instead it's added Decision reviews and appeals after Home. I think this is because we have this article listed under "More resources" on the decision reviews landing page. I can't figure out if there's a way in Drupal to keep the article linked on that landing page but not have that extra step show up in the breadcrumbs.

- [prod.cms.va.gov/resources/choosing-a-decision-review-option](http://prod.cms.va.gov/resources/choosing-a-decision-review-option)
- https://www.va.gov/resources/choosing-a-decision-review-option/

Randi wasn't expecting "decision reviews and appeals" to show up in the breadcrumbs. It should be Home > Resources and support > Choosing a decision review. R&S shouldn't be a category within decision reviews and appeals

Randi suspects that it's showing up that way because SWS has it as a link on the decision reviews landing page. But maybe it's something else?

## Additional context
This work may blur the lines between Drupal & Front-end work. We know Drupal generates some breadcrumbs, from menu or from path. We believe some page breadcrumbs are custom in the front-end. This ticket should take into account both backend/frontend breadcrumb handling.

## AC / Expected behavior
- [ ] Document any findings about how breadcrumbs work in comments
- [ ] Understand why this is happening
- [ ] Determine how widespread this particular issue is
- [ ] Meet with team/Dave to discuss next steps in context of all the breadcrumb woes

## Screenshots

<img width="777" alt="Screen Shot 2022-08-16 at 10 58 33 AM" src="https://user-images.githubusercontent.com/85581471/185191277-456b66c1-a997-4820-ba02-20813f703627.png">
<img width="665" alt="Screen Shot 2022-08-16 at 11 22 15 AM" src="https://user-images.githubusercontent.com/85581471/185191439-bcb14973-26bc-4c15-a580-4e8d70f609bd.png">

 ## Additional context
Resources and Support is currently disabled as a top-level menu item, with nothing listed beneath it.

## Labels
(You can delete this section once it's complete)
- [x] Issue type (red) (defaults to "Defect")
- [ ] CMS subsystem (green)
- [ ] CMS practice area (blue)
- [x] CMS workstream (orange) (not needed for bug tickets)
- [ ] CMS-supported product (black)

### CMS Team
Please check the team(s) that will do this work.

- [ ] `Program`
- [ ] `Platform CMS Team`
- [ ] `Sitewide Crew`
- [ ] `⭐️ Sitewide CMS`
- [X] `⭐️ Public Websites`
- [ ] `⭐️ Facilities`
- [ ] `⭐️ User support`

**Comment by allisonlu** [comment]
# Breadcrumbs... why! 🍞

## Notes on CMS backend
- Breadcrumbs are rendering similarly incorrectly in the CMS... this leads me to think this may be a backend issue
![Screenshot of breadcrumb bug on CMS backend](https://user-images.githubusercontent.com/8542413/188974151-d99c860e-95c6-4733-b811-ec8603b2d47b.png)

In the case it's NOT a backend issue, then continue on my liege...

- "Decision reviews and appeals" links to https://www.va.gov/decision-reviews/ (owned by VBA)

- ... while "[Choosing a decision review option](https://www.va.gov/resources/choosing-a-decision-review-option/)" (the page that we're investigating) is owned by OCTO Sitewide content

- In relation to [Decision reviews and appeals](https://www.va.gov/decision-reviews/) page, our page ["[Choosing a decision review option](https://www.va.gov/resources/choosing-a-decision-review-option/)"] contains:
  - **four** Link Teasers under Related information
  - **one** VA Benefit Hub (set to VA Decision reviews and appeals)
  - **one** Benefit Hub contact (set to VA decision reviews and appeals) under Contact information
  - **one** Primary category (set to Decision reviews and appeals) under Article categories 
   
- there might be logic in the breadcrumb template that's pulling these links into consideration

----

## While investigating 🔍...

- Changing the **Primary category** to anything besides the current **Decision reviews and appeals** on our page on [dev.cms.va.gov](https://dev.cms.va.gov/resources/choosing-a-decision-review-option) does **NOT** affect the breadcrumbs

- Changing the **Benefit Hub** to anything besides the current **Decision reviews and appeals** on our page on [dev.cms.va.gov](https://dev.cms.va.gov/resources/choosing-a-decision-review-option) does **NOT** affect the breadcrumbs

----

## (Possibly) relevant files

### `va.gov-cms`

- **docroot/modules/custom/va_gov_workflow_assignments/src/Plugin/Block/EntityMetaDisplay.php**

```
  public function getSectionHierarchyBreadcrumbLinks(NodeInterface $node) : string {
    $links = [];
    /** @var \Drupal\Core\Field\EntityReferenceFieldItemListInterface $field_administration */
    $field_administration = $node->get('field_administration');
    $owner_term = $field_administration->referencedEntities() ?
      $field_administration->referencedEntities()[0] : [];

    if (!empty($owner_term)) {
      $links[] = $this->getTermLink($owner_term);
      $parents = array_values($this->entityTypeManager
        ->getStorage('taxonomy_term')
        ->loadAllParents($owner_term->id())
      );

      for ($i = 1; $i < 3; $i++) {
        if (!empty($parents[$i])) {
          array_unshift($links, $this->getTermLink($parents[$i]));
        }
      }
    }

    return implode(' » ', $links);
  }

```

- **web/src/site/filters/liquid.js**
  - note this is very similar to the same code snippet in `content-build/src/site/filters/liquid.js`

```
  liquid.filters.deriveLcBreadcrumbs = (
    breadcrumbs,
    string,
    currentPath,
    pageTitle,
  ) => {
    // Remove any resources crumb - we don't want the drupal page title.
    const filteredCrumbs = breadcrumbs.filter(
      crumb => crumb.url.path !== '/resources',
    );
    // Add the resources crumb with the correct crumb title.
    filteredCrumbs.push({
      url: { path: '/resources', routed: false },
      text: 'Resources and support',
    });

    if (pageTitle) {
      filteredCrumbs.push({
        url: { path: currentPath, routed: true },
        text: string,
      });
    }

    return filteredCrumbs;
  };

```

- **web/src/site/includes/breadcrumbs.drupal.liquid**

----

### `content-build`

- **src/site/layouts/support_resources_detail_page.drupal.liquid**

```
{% include "src/site/includes/breadcrumbs.drupal.liquid" with constructLcBreadcrumbs = true titleInclude = true %}
```

- **src/site/includes/breadcrumbs.drupal.liquid**
  - everything in this file is pretty relevant

- **src/site/filters/liquid.js**
```
  liquid.filters.deriveLastBreadcrumbFromPath = (
    breadcrumbs,
    string,
    currentPath,
    replaceLastItem = false,
  ) => {
    const last = {
      url: { path: currentPath, routed: true },
      text: string,
    };

    if (replaceLastItem) {
      // replace last item in breadcrumbs with "last"
      breadcrumbs.splice(breadcrumbs.length - 1, 1, last);
    } else {
      breadcrumbs.push(last);
    }

    return breadcrumbs;
  };

  liquid.filters.deriveLcBreadcrumbs = (
    breadcrumbs,
    string,
    currentPath,
    pageTitle,
  ) => {
    // Remove any resources crumb - we don't want the drupal page title.
    const filteredCrumbs = breadcrumbs.filter(
      crumb => crumb.url.path !== '/resources',
    );
    // Add the resources crumb with the correct crumb title.
    filteredCrumbs.push({
      url: { path: '/resources', routed: false },
      text: 'Resources and support',
    });

    if (pageTitle) {
      filteredCrumbs.push({
        url: { path: currentPath, routed: true },
        text: string,
      });
    }

    return filteredCrumbs;
  };
```

### Resources

- Shopify - Liquid filters: https://shopify.dev/api/liquid/filters

- Shopify - Liquid template languange: https://shopify.github.io/liquid/

- `content-build` Guide to templates: https://github.com/department-of-veterans-affairs/content-build/tree/main/src/site

**Comment by jilladams** [comment]
Next step: Backend SPIKE to rule out CMS behavior affecting this. Closing.

**Comment by chri5tia** [comment]
The node section was defined incorrectly. The breadcrumbs are fixed when putting it in "--review decisions". As that is what the article topic is defined as and the node is part of benefit hubs, not resources and support. See https://github.com/department-of-veterans-affairs/va.gov-cms/issues/10680

**Comment by swirtSJW** [comment]
https://dsva.slack.com/archives/CDHBKAL9W/p1664973339806309?thread_ts=1660661901.076319&cid=CDHBKAL9W

TLDR:  Section (field_administration) does not control breadcrumbs (other than in VAMC Lovell),

**Comment by swirtSJW** [comment]
The "relevant file" docroot/modules/custom/va_gov_workflow_assignments/src/Plugin/Block/EntityMetaDisplay.php is misleading because it uses the word breadcrumb and is using field_administration, but that is for the "owner breadcrumb" not the page breadcrumb
![image](https://user-images.githubusercontent.com/5752113/194077392-9040ad1b-48fa-42be-a919-25cb0eabf6e2.png)
